### ISSUE TITLE
todo 

### ISSUE THREAD

**Issue by zdx3578** [createamind/cpc-pred-gan#1]
paper： better prediction imagination condition gan ： pattern gan；

学习规律的泛化。基于规律认识的视频预测。学规则，学规律？ 用错了数据集在视频预测中？

tech：
--1 more pred item; --1.1 cpc train item pred4 only gen1 1.1.1 cpc train 4-4 val 4-4 **and ！4-1 ！！**
--1.1.1  train cpc 4-4 gen 4-1 这个现在就可以！--1.2 **gen  no noise** is better??
--2 gen loss: only cpc loss or what weight??  or   看结果，loss比重一样也行，cpc收敛很快;还是 cpc 100吧
--**3 trained weight load;** keras tf ok；--3.2 print x item； 可以显示生成数字的cpc判别结果，如果不对再生成！

4 现在训练慢是训练数据是实时生成的，不是预生成的，应该生成好，直接训练
--**5 cpc和gan 循环训练，cpc的强化也继续，但是不是用gan的数据训练cpc，是原始cpc的数据训练cpc；gan的训练，Dcpc和Dcirit权重也需要动态变化，先生成好图片，在cpcloss生成准确数字的图片**
生成数字的预训练，降低训练生成数字对数字序列的影响。

5 64X64  大数字。

app：
**0 not mnist ; 标准数字的数据集；学规律，不是学数字识别**
1 more practice; 1.1 123 124; just train ok； 1-500  600-700valid not work；

1.2  1+3=4;  9内加法99内加法
数据是否用标准数字图像，而不是mnist数据集？
1.1.1  数据集生成；
**基于之前1234最基本的训练基础上再训练，不是直接记忆序列，从规律开始学习。
个位数训练的顺序情况应用到加1的加法上去；同时训练加法和顺利排列？？
或者两种训练同时学习，rnn如何兼顾两种格式？？ 即学习个位数的循环排列顺序，也学习数字增长的规律，
或 oneshot；每种数据给一点提示。，数据分段生成。**

**condition gan ref cgan**

1.2.1 数据集生成；

？1 进位规律的学习。

-------------------------------------------------------------------------

2 picture and video and carla  
U-net pix2pix 
c3d   cdu-net;  vid2vid

学规则，学规律？ 用错了数据集？

3 video+action ；红绿灯问题
cpchrl
cpc(video+action::video+action paire)

**4 rl；强化学习只需要cpc的预测隐变量即可，隐变量生成再可视化跟人交互，**
4.2： rl涉及动作。 根据规律，且加入动作进行预测。 规律+动作conditon规律的组合生成隐变量
rnn的数字生成器，生成多个的序列，根据序列个数生成可变序列， rnn的condtion gan；

**Comment by zdx3578** [comment]
-        terms=4,
+        terms=1,  change
gan train error！

**Comment by zdx3578** [comment]
预测多个，生成一个：
![image](https://user-images.githubusercontent.com/382863/47057914-58189980-d1f5-11e8-8992-bceb5b305c12.png)
当前最好效果的配置
loss
![image](https://user-images.githubusercontent.com/382863/47138938-3dc1e700-d2ed-11e8-99a3-f1df2be864a5.png)

**Comment by zdx3578** [comment]
pred more：

sdc       8688  8451 99 11:24 pts/29   05:00:05 python train_model.py --name genmore1weight1 --cpc-epochs=51 --gan-weight=0.2
sdc      15907 27403 99 13:52 pts/30   02:03:23 python train_model.py --name genmore1weight2ncritic1pred4 --cpc-epochs=51 --gan-weight=2 --predict-terms=4 --terms=4

**Comment by zdx3578** [comment]
1234 1235 1236 train converge:

![image](https://user-images.githubusercontent.com/382863/47218770-780bb100-d3df-11e8-81c8-4f3274817148.png)

**Comment by zdx3578** [comment]
![image](https://user-images.githubusercontent.com/382863/47248517-825c9800-d43d-11e8-94fd-c52b3b38ed13.png)

**Comment by zdx3578** [comment]
tf版本的cpc train 1234 1235 貌似收敛效果比之前的keras要好一些，keras版本好像只是到99.9几的准确率
![image](https://user-images.githubusercontent.com/382863/47277430-12900e00-d5f3-11e8-806e-c1e2934cf71d.png)
